### DISCUSSION TITLE
How to open a URL in the GitHub app

### DISCUSSION THREAD

**Discussion by jidanni** [community/community#124987] [Mobile] [answered]
Let's say I'm not using the app and I run across a GitHub URL, 
https://github.com/python-mechanize/mechanize/

How do I get it to open inside the GitHub app, and not just inside a browser? 

If I share it into the GitHub app, the GitHub app thinks I want to create a new issue, and asks me what repository I wish to create that issue in.

One would think that most browsers would recognize it is a GitHub URL, and open it in the GitHub app on the user's cell phone. 

But that doesn't happen for me, despite my best efforts in my browser's settings, and the same thing in various other browsers too.

Sure, I could just fire up the GitHub app and paste that URL into some issue in my private repository and then tap that rendered URL in the issue and that would get me to where I want to go. Just like you can tap it right above if you're reading this in the GitHub app. But I'm talking about the case where I'm not inside the GitHub app yet.

**Comment by gitstua** [comment]
Which mobile type are you using @jidanni? In `iOS` this is working for me by default but [this](https://www.reddit.com/r/ios/comments/rxyj3r/comment/jetycls/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) appears to be how to force it in iOS 

For `Android` [this](https://www.reddit.com/r/Android/comments/qrxsh7/guide_how_to_fix_horrible_handling_of_links_on/) may help

**Reply by jidanni** [reply]
Android 13. Thanks. No wonder.
Say, isn't GitHub a big corporation?
Well, from what I read in your links it seems that they ought to be able
to force using the GitHub app to open GitHub links,
like the other big companies mentioned.

**Comment by Janethq** [comment]
There is no direct way to open a GitHub URL in the GitHub app from a browser on your mobile device. This is because the GitHub app does not register itself as the default handler for GitHub URLs on mobile devices.

However, there are a few workarounds you can try:

1. Copy the GitHub URL and then open the GitHub app on your mobile device. Once the app is open, you can paste the URL into the search bar at the top of the screen and press enter. This should take you to the repository or file you were trying to view.

2. If you have the GitHub app installed on your mobile device, you can try adding the github-protocol handler to your device's custom URL scheme. This will allow you to open GitHub URLs directly in the GitHub app by adding github- before the https:// part of the URL. For example, the URL https://github.com/python-mechanize/mechanize/ would become github-https://github.com/python-mechanize/mechanize/. Note that this method may not work on all devices or versions of the GitHub app.

3. If you are using an iOS device, you can try using the "Open in" feature in your browser to open the GitHub URL in the GitHub app. To do this, tap the "Share" button in your browser and select "Open in" from the menu. Then, select the GitHub app from the list of available apps.

**Reply by jidanni** [reply]
> This is because the GitHub app does not register itself as the default handler for GitHub URLs on mobile devices.

Ah, so it's not the users' fault in the first place. Ah ha!

Any special reason why the GitHub company would go through all the trouble to create an app, and then not register it properly? Perhaps they aren't finished testing it yet?

**Reply by jidanni** [reply]
By the way, entering the URL in the search box didn't work. See https://github.com/orgs/community/discussions/125564 .

**Reply by queenofcorgis** [reply]
@Janethq 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by elisealix22** [comment, marked as answer]
@jidanni GitHub links are supported by the Android app, however its possible either your system settings and/or browser of choice disable this. I would recommend first checking system settings in `Settings -> Apps -> GitHub` and ensure the app is allowed to open supported links.

If that doesn't work or is already enabled, some browsers have custom settings to allow app links to open outside the browser. These settings may be disabled by default, and need to be actively turned on. Open your default/preferred browser. Scroll through the available settings to see if there is a toggle to open supported links. The process may vary depending on the browser you're using.

For example:
- In Firefox, this setting is under Advanced > Open links in apps > Ask before opening.
- In Brave, this setting is in Brave Shields & Privacy > Allow app links to open in apps outside of Brave.

Changing this setting to ask before opening will prompt for your permission before it opens a link in an app. When a dialog box appears, asking if you want to open the link in the GitHub Mobile app or stay in the browser, choosing to stay in the browser will avoid a redirection loop.

If those two suggestions don't work, please let us know which browser you are using by default and we will try to reproduce locally!

**Reply by jidanni** [reply]
Unfortunately I'm using this other browser called Chrome. 

I suppose they ripped out that choice, because it's in Brave but not in Chrome.

<Del>The only other thing I could think of, is I indeed have set the GitHub app to be a "deep sleeping" app  on Samsung. </Del>
